data_analysis/plots.py

```diff
@@ -1035,16 +1052,23 @@ def run_mvp() -> None:
     )
 
 
-def main() -> None:
-    """Generate maps of all routes, top 10 best routes,
+def main(day_type: str = None) -> None:
+    """ Generate maps of all routes, top 10 best routes,
     top 10 worst routes, and ridership
+
+    Args:
+        day_type (str, optional): day_type to filter by. Defaults to None.
     """
     logger.info("Creating GeoDataFrame")
     gdf = static_gtfs_analysis.main()
 
     logger.info("Getting latest real-time and schedule comparison data")
 
     combined_long_df, summary_df = compare_scheduled_and_rt.main(freq='D')
+
+    if day_type is not None:
+        summary_df = filter_day_type(summary_df, day_type=day_type)
+
     summary_df = calculate_percentile_and_rank(summary_df, col="ratio")
```

Comment:
Do we need to calculate summary_df here after just calculating it on line 1070?

Reply:
`summary_df` is just being modified after it's passed to `filter_day_type` and `calculate_percentile_and_rank`. It's calculated on line 1067.